### PR TITLE
Trending developers

### PR DESCRIPTION
**Pull request by gabrieledarrigo**
### DESCRIPTION

Added a component that renders the daily list of trending developers; when a user clicks on one of the links the application should render the related developer's card.

#### Screenshot

image

### CODE REVIEW

File: src/components/trending-developers/TrendingDevelopers.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }, []);
+
+  const goTo = useCallback((username) => (e) => {
+    e.preventDefault();
+    history.push(`/${username}`);
+  }, [history]);
+
+  return (
+    <div className={styles.trendingDevelopers}>
+      {trendingDevelopers && trendingDevelopers.length > 0 && (
```
Comment (ohelixa) [inline src/components/trending-developers/TrendingDevelopers.jsx:31]: We might start using an util like: `isEmpty()`? In order to use it here like so: !isEmpty(trendingDevelopers)

File: src/components/trending-developers/TrendingDevelopers.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                onClick={goTo(username)}
+              >
+                {username}
+              </a>
+            </li>
+          ))}
+        </ul>
+      )}
+
+      {trendingDevelopers && trendingDevelopers.length === 0 && (
```
Comment (ohelixa) [inline src/components/trending-developers/TrendingDevelopers.jsx:52]: Here as: isEmpty(trendingDevelopers)

File: src/components/trending-developers/TrendingDevelopers.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const history = useHistory();
+  const [trendingDevelopers, setTrendingDevelopers] = useState();
+
+  useEffect(() => {
+    async function fetch() {
+      try {
+        const result = await getTrendingDevelopers();
+        setTrendingDevelopers(result);
+      } catch (err) {
+        // Fail silently
```
Comment (ohelixa) [inline src/components/trending-developers/TrendingDevelopers.jsx:16]: What about handling the error with a state?

File: src/components/trending-developers/TrendingDevelopers.jsx (hunk 1 above)
Comment (ohelixa) [reply src/components/trending-developers/TrendingDevelopers.jsx:31]: What about adding: https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining ?

File: src/App.jsx [hunk 4]
```diff
@@ -76,6 +77,16 @@ function App({
         )}
       </div>
 
+      <header className={styles.header}>
```
Comment (ohelixa) [inline src/App.jsx:80]: This header section should be part of the `TrendingDevelopers` component

File: src/components/trending-developers/TrendingDevelopers.jsx (hunk 1 above)
Comment (gabrieledarrigo) [reply src/components/trending-developers/TrendingDevelopers.jsx:31]: I like your `isEmpty` check; 
In regards to optional chaining and null coalescing operator, yeah, why not, but I'll leave this to you, maybe a PR? : )

File: src/components/trending-developers/TrendingDevelopers.jsx (hunk 2 above)
Comment (gabrieledarrigo) [reply src/components/trending-developers/TrendingDevelopers.jsx:52]: That's fine 👍

File: src/components/trending-developers/TrendingDevelopers.jsx (hunk 3 above)
Comment (gabrieledarrigo) [reply src/components/trending-developers/TrendingDevelopers.jsx:16]: Yeah, I thought about this, but the component renders a list, so I'm uncertain if rendering an error to the user is the right UI pattern; that's why I merged the "can't fetch data" and "trending developers is empty" errors:

https://github.com/gabrieledarrigo/github-cards/pull/3/files#diff-94f6d0873bc97d5fe79aa64a8505e5a9R54

Don't know, have you got better ideas?

File: src/App.jsx (hunk 4 above)
Comment (gabrieledarrigo) [reply src/App.jsx:80]: Yep, I guess you're right!

File: src/components/trending-developers/TrendingDevelopers.jsx (hunk 3 above)
Comment (ohelixa) [reply src/components/trending-developers/TrendingDevelopers.jsx:16]: Uhm, that's also true. We can leave it like this for now.

Comment (gabrieledarrigo) [reply src/components/trending-developers/TrendingDevelopers.jsx:16]: Ok, you are welcome!

**Pull request merged**